### ISSUE TITLE
Unable to share vehicle

### ISSUE THREAD

**Issue by ettiennegous** [iainbullock/tesla-http-proxy-docker#41]
I have stepped through the process 3 times now and keep getting the below error after I scan the QR code on the tesla website and it launches the Tesla Phone App.

I have no idea whats wrong and would appreciate any ideas / feedback on how to troubleshoot

![image](https://github.com/iainbullock/tesla-http-proxy-docker/assets/556767/af9a59bd-0a5d-4c91-bbd7-5dcf2482cb34)

## Things I have tried
1. Went over the docs multiple times and ensured all the steps work one after the other for the initial configuration
1. I have created a new app 3 times and stepped through the things methodically
1. I am certain the public key `com.tesla.3p.public-key.pem` is available on the FQDN 

Here is a redacted log output that may help with investigation

```
Configuration options not set in /data/config.sh, exiting
Configuration Options are:
CLIENT_ID=<redacted>
CLIENT_SECRET=<redacted>
DOMAIN=tesla.<redacted>.website
PROXY_HOST=teslaproxy.local
REGION=North America, Asia-Pacific
Setting up for first launch
Setting up GnuPG and password-store
gpg: keybox '/data/gnugpg/pubring.kbx' created
gpg: /data/gnugpg/trustdb.gpg: trustdb created
gpg: directory '/data/gnugpg/openpgp-revocs.d' created
gpg: revocation certificate stored as '/data/gnugpg/openpgp-revocs.d/<redacted>.rev'
gpg: checking the trustdb
gpg: marginals needed: 3  completes needed: 1  trust model: pgp
gpg: depth: 0  valid:   1  signed:   0  trust: 
gpg: next trustdb check due at 2027-04-26
/data/gnugpg/pubring.kbx
------------------------
pub   ed25519 2024-04-26 [SC] [expires: 2027-04-26]
      <redacted>
uid           [ultimate] myself
sub   cv25519 2024-04-26 [E]
created directory: '/data/password-store/'
Password store initialized for myself
Generating keypair
-----BEGIN PUBLIC KEY-----
<redacted>
-----END PUBLIC KEY-----
SSL certificate does not exist
Generating self-signed SSL certificate
-----
Starting temporary Python app for authentication flow. Delete /data/access_token to force regeneration in the future
[13:46:00] main:INFO: *** Starting Flask server... ***
[13:46:00] werkzeug:INFO: WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:8099
 * Running on http://172.20.0.17:8099
[13:46:00] werkzeug:INFO: Press CTRL+C to quit
[13:46:03] werkzeug:INFO: 172.20.0.1 - - [26/Apr/2024 13:46:03] "GET / HTTP/1.1" 200 -
[13:46:29] main:INFO: callback args: ImmutableMultiDict([('locale', 'en-US'), ('code', '<redacted>'), ('state', '<redacted>'), ('issuer', 'https://auth.tesla.com/oauth2/v3')])
[13:46:29] main:INFO: Info to enter into Tesla Custom component:
Refresh token  : <redacted>
Proxy URL      : https://teslaproxy.local:4430
SSL certificate: /share/home-assistant/selfsigned.pem
Client ID      : <redacted>
[13:46:29] werkzeug:INFO: 172.20.0.1 - - [26/Apr/2024 13:46:29] "GET /callback?locale=en-US&code=<redacted>&state=<redacted>&issuer=https://auth.tesla.com/oauth2/v3 HTTP/1.1" 200 -
[13:46:34] main:INFO: *** Generating Partner Authentication Token ***
[13:46:34] main:INFO: {"access_token":"<redacted>","expires_in":28800,"token_type":"Bearer"}
[13:46:34] main:INFO: *** Registering Tesla account ***
[13:46:36] main:INFO: {"response":{"client_id":"<redacted>","name":"Tesla1 Home Assistant","description":"Home Assistant Integration for the Tesla Model Y","domain":"tesla1.<redacted>.website","ca":null,"created_at":"2024-04-26T13:46:36.182Z","updated_at":"2024-04-26T13:46:36.182Z","enterprise_tier":"free","account_id":"<redacted>","issuer":null,"csr":null,"csr_updated_at":null,"public_key":"<redacted>"}}
[13:46:36] werkzeug:INFO: 172.20.0.1 - - [26/Apr/2024 13:46:36] "GET /register-partner-account HTTP/1.1" 302 -
[13:46:36] werkzeug:INFO: 172.20.0.1 - - [26/Apr/2024 13:46:36] "GET /?success=1 HTTP/1.1" 200 -
```

**Comment by ettiennegous** [comment]
As a side note does anyone know how to delete an app from their tesla developer dashboard?  
I suspect its not possible?

**Comment by iainbullock** [comment]
Let me know how you got on. Can I close this Issue?

**Comment by ettiennegous** [comment]
I  read through the other threads @iainbullock with no luck on their suggestions. 
I decided to leave it for a while and give it a go again, I started from scratch and had the exact same outcome.
My next move is going to be to play with the fleet telemetry tools and instructions from
https://github.com/teslamotors/fleet-telemetry

Given no one else has reported this issue it seems to be isolated to myself.  
Feel free to close
